### PR TITLE
SelectionZone: Document component interface, props, and remove any type usage

### LINKED ISSUE #8808: DetailsList set selection with keyboard
My DetailsList needs to be keyboard accessible. I am using selection={this._selection} to manage selection with mouse clicks and that is working correctly. I can use the up and down arrow keys to move focus between the items in the list but pressing space and enter does not fire the selection event. I have tried using onItemInvoked which fires correctly on Enter, but there is no way to set the Selection object's selected value. It won't update the 'selected' item's styles. 

I need to make this component keyboard accessible but I'm not sure how to make the Selection object work for both mouse click and keyboard click.

### PR DESCRIPTION
**Pull request by KevinTCoughlin**
#### Pull request checklist

- [x] Addresses an existing issue: Fixes #8808
- [x] Include a change request file using `$ npm run change`

Related: #8808 

#### Description of changes

It is often unclear to developers on how / when to use `SelectionZone` and its related behaviors. This PR:

- Adds documentation to the exported interfaces
- Removes `any` type usage in private method
- Consolidates `ev.target` as a local variable.

Prior to #8808, I had no idea that `isSelectedOnFocus` prop was an option for `DetailsList` behavior customization.

#### Focus areas to test

- None.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/OfficeDev/office-ui-fabric-react/pull/8864)

### CODE REVIEW

**Comment by KevinTCoughlin** [comment]
I'm going to opt for `fabric-7` as the target for safety, will re-open.

File: apps/a11y-tests/src/tests/__snapshots__/Button.test.ts.snap [hunk 1]
```diff
@@ -4,33 +4,11 @@ exports[`runs Axe on Button 1`] = `
...
-            "snippet": Object {
-              "text": "<div data-automation-id=\\"test\\" aria-roledescription=\\"split button\\" aria-disabled=\\"false\\" data-is-focusable=\\"true\\" role=\\"button\\" aria-haspopup=\\"true\\" aria-expanded=\\"false\\" class=\\"css-260\\" tabindex=\\"0\\">",
-            },
-          },
-        ],
-        "fullyQualifiedLogicalName": "div:nth-child(1) > .css-260[aria-roledescription=\\"split\\\\ button\\"][aria-haspopup=\\"true\\"]",
-      },
-    ],
-=======
->>>>>>> upstream/master
```
Comment (Vitalius1) [inline apps/a11y-tests/src/tests/__snapshots__/Button.test.ts.snap]: What is up with this file coming from the `a11y-tests` app?

Comment (KevinTCoughlin) [reply apps/a11y-tests/src/tests/__snapshots__/Button.test.ts.snap]: Good question!

**Comment by KevinTCoughlin** [comment]
Idk what's going on with this branch.. I'll take another pass at it. Seems like there's some bad test in `fabric-7`.

**Pull request merged**